reproject/healpix.py

```diff
@@ -0,0 +1,213 @@
+
+    return new_hdu
+
+def healpix_hdu_to_hdu(hp_hdu, reference_header, field=None, **kwargs):
+    """
+    Convert a HEALPIX binary-table HDU to a target FITS header.
+
+    Parameters
+    ----------
+    hp_hdu : fits.BinTableHDU
+        The HDU containing the healpix data table
+    reference_header : fits.Header
+        The target header, containing a valid WCS
+    field : None or string
+        The field name containing the data to be reprojected.  If not
+        specifies, defaults to the first field in the BinTable
+        
+    Returns
+    -------
+    fits.PrimaryHDU containing the reprojected image
+    """
+    if field is None:
+        field = hp_hdu.data.columns[0].name
+
+    nested = hp_hdu.header['ORDERING'] == 'NESTED'
+    coordsys = hp_hdu.header['COORDSYS'].lower()
+
+    if coordsys not in valid_coordinate_systems:
+        raise KeyError("Coordinate system was %s, which is not one of %s" % (coordsys,valid_coordinate_systems))
+
+    healpix_data = hp_hdu.data[field]
+
+    reprojected_data = healpix_to_image(healpix_data, reference_header,
+                                        coordsys, nest=nested, **kwargs)
+
+    new_hdu = fits.PrimaryHDU(data=reprojected_data, header=reference_header)
+
+    return new_hdu
+
+def healpix_to_image(healpix_data, reference_header, hpx_coord_system,
+                     interp=True, nest=False):
+    """
+    Convert image in HEALPIX format to a normal FITS projection image (e.g.
+    CAR or AIT).
+
+    Parameters
+    ----------
+    healpix_data : `numpy.ndarray`
+        HEALPIX data array
+    reference_image : `astropy.io.fits.ImageHDU`
+        A reference image to project to.  Must have a 'COORDSYS' keyword of
+        either 'galactic' or 'icrs'
+    hpx_coord_system : 'galactic' or 'icrs'
+        The target coordinate system.  Should be derived from the HEALPIX
+        COORDSYS keyword if it is a FITS file
+    nest : bool
+        The order of the healpix_data, either nested or ring.  Stored in 
+        FITS headers in the ORDERING keyword.
+    interp : bool
+        Get the bilinear interpolated data?  If not, returns a set of neighbors
+
+    Returns
+    -------
+    reprojected_data : `numpy.ndarray`
+        HEALPIX image resampled onto the reference image
+    
+    Examples
+    --------
+    >>> import healpy as hp
+    >>> from astropy.io import fits
+    >>> from reproject.healpix import healpix_to_image
+    >>> healpix_filename = 'healpix.fits'
+    >>> healpix_data = hp.read_map(healpix_filename)
+    >>> healpix_system = fits.getheader(healpix_filename,ext=1)['COORDSYS']
+    >>> healpix_isnested = fits.getheader(healpix_filename,ext=1)['ORDERING'] == 'NESTED'
+    >>> reference_image = fits.open('reference_image.fits')[0]
+    >>> reprojected_data = healpix_to_image(healpix_data, reference_image, healpix_system, nest=healpix_isnested)
+    >>> fits.writeto('new_image.fits', reprojected_data, reference_image.header)
```

Comment:
Problem here: by default, `hp.read_map` reorders the pixels to `RING` order. I'll make a pull request to fix this...